internal/inspect/inspect.go

```diff
@@ -1,42 +1,163 @@
 package inspect
 
 import (
+	"bytes"
 	"fmt"
 	"io"
 	"sort"
+	"strings"
 	"text/tabwriter"
 
+	"github.com/docker/app/render"
 	"github.com/docker/app/types"
+	"github.com/docker/app/types/settings"
+	composetypes "github.com/docker/cli/cli/compose/types"
 )
 
 // Inspect dumps the metadata of an app
-func Inspect(out io.Writer, app *types.App) error {
+func Inspect(out io.Writer, app *types.App, argSettings map[string]string) error {
+	// Render the compose file
+	config, err := render.Render(app, argSettings)
+	if err != nil {
+		return err
+	}
+
+	// Extract all the settings
+	settingsKeys, allSettings, err := extractSettings(app, argSettings)
+	if err != nil {
+		return err
+	}
+
+	var sections []*bytes.Buffer
+
+	// Add Meta data
+	addMetadata(&sections, app)
```

Comment:
Why not `printing` those instead of putting them in a `[]*bytes.Buffer` ? :thinking:
It would make the code easier :angel:

Reply:
Well, all the sections (except metadata) are optional (if they are empty). Then I have a final `\n` problem. Inside each section I can't know if it's the last one or not, and if I need to print a `\n`. That's why I use a `strings.Join` at the end, and print everything at the end.

Reply:
:thinking: You *can* know if each section is printing a `\n` before anything (thus you would not have that problem)

Reply:
This solution is too simple, I prefer mine :trollface: